core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1295,4 +1295,10 @@ private static SqlDataTypeSpec createSqlDataTypeSpecByName(String typeAlias,
         typeAlias, typeName, SqlParserPos.ZERO);
     return new SqlDataTypeSpec(typeNameSpec, SqlParserPos.ZERO);
   }
+  @Override public void quoteStringLiteral(StringBuilder buf, String charsetName,
+      String val) {
+    buf.append(literalQuoteString);
+    buf.append(val.replace(literalEndQuoteString, literalEscapedQuote));
```

Comment:
something does not look right here.
@kk19ag @praveendatametica

Reply:
Hey Abbas, do we update the Default implementation in SqlDialect (calcite) which currently says that it will convert non Ascii characters to UNICODE. Also if charset is specified then it would append the charset name after the literal... This default behaviour was also happening earlier. We string replace that in TranslatorUtil as of now. 

So can we say we will have a different version of SqlDialect.quoteStringLiteral() than calcite? because these fail safes makes no sense to us or may be they do if target doesnt support some specific charset.

@abbas-gadhia @saurabhgupta-datametica @purvalila @abhisheikpushp

Reply:
we should start a discussion on the ML. If the target supports it, i dont see why we can't change the code in the main sqldialect class

Reply:
@abbas-gadhia but they might have their own through process around the way they treat literals as default.....
we could pick their brains but i think they would have their reasons not to budge.
@saurabhgupta-datametica please start a thread.

Reply:
@abbas-gadhia @saurabhgupta-datametica any updates on this we need this fix.